doc/ecosystem.md

```diff
@@ -0,0 +1,131 @@
+# Ecosystem
+
+PyGMT provides a Python interface to the Generic Mapping Tools (GMT), which is a command
+line program that provides a wide range of tools for manipulating geospatial data and
+making publication-quality maps and figures. PyGMT integrates well with the
+[scientific Python ecosystem](https://scientific-python.org/), with [NumPy][] for its
+fundamental array data structure, [pandas][] for tabular data I/O and [Xarray][] for
+raster grids/images/cubes I/O.
+
+In addition to these core dependencies, PyGMT also relies on several optional packages to
+provide additional functionality for users.
+
+*This page was adapted from [GeoPandas's Ecosystem](https://geopandas.org/en/latest/community/ecosystem.html) page.*
+
+## PyGMT dependencies
+
+### Required dependencies
+
+#### NumPy
+
+[NumPy][] is the fundamental package for scientific computing in Python. It is a Python
+library that provides a multidimensional array object, various derived objects (such as
+masked arrays and matrices), and an assortment of routines for fast operations on arrays,
+including mathematical, logical, shape manipulation, sorting, selecting, I/O, discrete
+Fourier transforms, basic linear algebra, basic statistical operations, random simulation
+and much more.
+
+#### pandas
+
+[pandas][] is a Python package providing fast, flexible, and expressive data structures
+designed to make working with "relational" or "labeled" data both easy and intuitive.
+It aims to be the fundamental high-level building block for doing practical, real-world
+data analysis in Python. Additionally, it has the broader goal of becoming the most
+powerful and flexible open source data analysis/manipulation tool available in any
+language. It is already well on its way toward this goal.
+
+#### Xarray
+
+[Xarray][] is an open source project and Python package that introduces labels in the
+form of dimensions, coordinates, and attributes on top of raw NumPy-like arrays, which
+allows for more intuitive, more concise, and less error-prone user experience.
+
+### Optional dependencies
+
+#### IPython
+
+[IPython][] provides a rich toolkit to help you make the most of using Python
```

Comment:
This paragraph comes from https://ipython.readthedocs.io/en/stable/#ipython-documentation